src/sapi.rs

```diff
@@ -326,48 +329,46 @@ fn test_format_duration() {
     assert_eq!(format_duration(duration), "0:00");
 }
 
-impl<'a> Windowed for SpVoice<'a> {
-    fn window_proc(
-        &mut self,
-        msg: minwindef::UINT,
-        w_param: minwindef::WPARAM,
-        l_param: minwindef::LPARAM,
-    ) -> Option<minwindef::LRESULT> {
+impl Windowed for SpVoice {
+    fn window_proc(&mut self, msg: u32, w_param: WPARAM, l_param: LPARAM) -> Option<LRESULT> {
         match msg {
-            winuser::WM_DESTROY | winuser::WM_QUERYENDSESSION | winuser::WM_ENDSESSION => close(),
+            wm::WM_DESTROY | wm::WM_QUERYENDSESSION | wm::WM_ENDSESSION => close(),
             WM_SAPI_EVENT => {
                 let status = self.get_status();
+                // convert rate from range (-10, 10) to (0, 20)
+                let rate_shifted = 10u32
+                    .checked_add_signed(self.get_rate())
+                    .expect("bad rate < -10") as usize;
```

Comment:
This replaces the `rate as usize +10` expression used later in this method, which was throwing an integer overflow error at runtime when `rate` was negative. This seems to be the right solution, which only complains if the resulting sum overflows (which should never happen).